app/src/main/java/de/westnordost/streetcomplete/quests/smoking/AddSmoking.kt

```diff
@@ -0,0 +1,77 @@
+package de.westnordost.streetcomplete.quests.smoking
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.isShopExpressionFragment
+import de.westnordost.streetcomplete.data.meta.updateWithCheckDate
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.filter
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.CITIZEN
+import de.westnordost.streetcomplete.ktx.arrayOfNotNull
+import de.westnordost.streetcomplete.ktx.containsAny
+
+class AddSmoking : OsmFilterQuestType<SmokingAllowed>() {
+
+    private val elementFilterBasic = """
+        (
+            amenity ~ bar|cafe|pub|biergarten|restaurant|food_court|nightclub|stripclub
+            or leisure ~ outdoor_seating
+            or (
+                (amenity ~ fast_food|ice_cream or shop ~ ice_cream|deli|bakery|coffee|tea|wine)
+                and (
+                    (outdoor_seating and outdoor_seating != no)
+                    or (indoor_seating and indoor_seating != no)
+                )
+            )
+        )
+    """
+
+    override val elementFilter = """
+         nodes, ways, relations with
+         ${elementFilterBasic}
+         and takeaway != only
+         and (!smoking or smoking older today -8 years)
+    """
+
+    override val changesetComment = "Add smoking status"
+    override val wikiLink = "Key:smoking"
+    override val icon = R.drawable.ic_quest_smoke
+    override val isReplaceShopEnabled = true
+    override val defaultDisabledMessage = R.string.default_disabled_msg_go_inside_regional_warning
+
+    override val questTypeAchievements = listOf(CITIZEN)
+
+    override fun getTitle(tags: Map<String, String>) =
+        if (hasProperName(tags))
+            R.string.quest_smoking_name_type_title
+        else
+            R.string.quest_smoking_no_name_title
+
+    override fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String> {
+        val name = tags["name"] ?: tags["brand"] ?: tags["operator"]
```

Comment:
I am not convinced that `operator` is valuable here with any conceivable correct tagging,

Reply:
Well they **_are_** on the map; check this overpass for example: https://overpass-turbo.eu/s/1gwj

Unfortunately timeouts on bigger areas, but for example in city of Rome, Italy (default overpass-turbo.eu location) it seems to be _at least_ as popular as places without `name` but with `brand` (and in this particular case much more popular; although at other places `brand` might be more popular; see also https://github.com/streetcomplete/StreetComplete/pull/3784#discussion_r815349445)

So I think it would be better to show it to the user, then to hide it (in cases `operator` exists, but `name` and `brand` do not). Or do you disagree?

Reply:
I disagree for two reasons:
1. AFAIR it is inconsistent with the rest of the code. There are many places where you find `tags["name"] ?: tags["brand"]`. If `operator` is going to be added as a second fallback, it should be done consistently everywhere (and not in this PR)
2. In my experience, some shops are unnamed in the sense that there is no proper sign with a name. But sometimes, the  owner of the shop is known. `operator` is in this case used for the owner of the shop. But this is usually not signed. Depending on the country and legislation, shop owners may be required to display their license somewhere visible (a piece of paper) and this is why the name of the owner may even be known. But this is not suitable for orientation and/or referring to the shop because it is a very non-prominent piece of information - if it is displayed at all

Reply:
1. Yes, I think it would be good fallback in the rest of the code too if neither `name` nor `brand` are present, but I do not presume to make a decision about that for whole codebase. Some other pieces of software like OsmAnd will display `operator` if `name` is missing, so it seems to be (at least partly) accepted practice.
2. Use case I've commonly seen is that cafe/fast_food is part of some supermarket, so while it is not named separately, it is operated by that supermarket. Also it might depend on cultural differences; some places especially smaller family restaurants, cafes etc. might take pride / be referred to by its proprietor.

Anyway, let me know of your decision, and I'll abide.

Reply:
Yes, please remove operator

Reply:
removed